x-pack/plugins/infra/public/containers/logs/log_entries/index.ts

```diff
@@ -124,48 +144,99 @@ const useFetchEntriesEffect = (
   dispatch: Dispatch,
   props: LogEntriesProps
 ) => {
-  const { getLogEntriesAround, getLogEntriesBefore, getLogEntriesAfter } = useGraphQLQueries();
-
-  const [prevParams, cachePrevParams] = useState(props);
+  const [prevParams, cachePrevParams] = useState<LogEntriesProps | undefined>();
   const [startedStreaming, setStartedStreaming] = useState(false);
 
-  const runFetchNewEntriesRequest = async (override = {}) => {
+  const runFetchNewEntriesRequest = async (overrides: Partial<LogEntriesProps> = {}) => {
+    if (!props.startTimestamp || !props.endTimestamp) {
+      return;
+    }
+
     dispatch({ type: Action.FetchingNewEntries });
+
     try {
-      const payload = await getLogEntriesAround({
-        ...omit(props, 'jumpToTargetPosition'),
-        ...override,
-      });
+      const fetchArgs: LogEntriesRequest = {
+        sourceId: overrides.sourceId || props.sourceId,
+        startTimestamp: overrides.startTimestamp || props.startTimestamp,
+        endTimestamp: overrides.endTimestamp || props.endTimestamp,
+        query: overrides.filterQuery || props.filterQuery || undefined, // FIXME
```

Comment:
Do you want to tackle that FIXME in a follow-up PR?

Reply:
ah! thanks for reminding me. I did write about it [here](https://github.com/elastic/kibana/pull/54280#discussion_r377085060)

> In the API query can be string or undefined. props.filterQuery can be string or null.
>
> I tend to prefer undefined over null (I know the argument that null explicitly "no value", but IMO that can also be represented with undefined with less gotchas (i.e. typeof null == 'object). I don't know what's the official Elastic take on this, so I haven't fixed it yet).

It's basically deciding between using `null` or `undefined` in the API definition. I lean towards the second and that's what I wrote originally in the API types. I don't know what's the Elastic Way™ or if there's any at all. The feeling I got is that `null` is more used but maybe we want to change that? (please :D?)

Reply:
The Elastic way is what we make of it :wink: I can't make an argument for `null` over `undefined` beyond what you already mentioned, so take your pick.